src/channels/ChallengeChannels/backgroundChallengeChannel.ts

```diff
@@ -19,10 +20,12 @@ export async function produceEncryptedChallenge(
 
   const { dAppEncryptionDidKey, sporranEncryptionDidKeyUri } = encryption;
 
-  const { sealed, nonce } = naclSeal(
+  const nonce = randomAsU8a(24);
+  const sealed = box(
     Utils.Crypto.coToUInt8(challenge),
-    encryption.encryptionKey.secretKey,
+    nonce,
     dAppEncryptionDidKey.publicKey,
+    encryption.encryptionKey.secretKey,
   );
```

Comment:
I wonder if this could be done with @noble/curves which polkadot uses now

Reply:
I'd suggest
```js
  const { box, nonce } = Utils.Crypto.encryptAsymmetric(
    Utils.Crypto.coToUInt8(challenge),
    dAppEncryptionDidKey.publicKey,
    encryption.encryptionKey.secretKey,
  );
```

Reply:
> I wonder if this could be done with @noble/curves which polkadot uses now

The tweetnacl / libsodium 'box' algorithm is a combination of a Diffie-Hellman key agreement on the ed25519 curve, then a salsa20 stream cipher encryption using the result of the key agreement, and finally the addition of a message authentication code (MAC) using that key. Noble only provides an implementation of the key agreement afaik

Reply:
I think they are just not using asymmetric encryption at all at this point. They only probably only used secretbox internally, likely for encrypting private keys at rest.